src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -12,19 +15,43 @@ public class Tag {
     public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
     public static final String VALIDATION_REGEX = "\\p{Alnum}+";
 
+    /*
+     * Hashmap for keeping track of unique tags
+     */
+    private static final Map<String, Tag> TAGS = new HashMap<>();
+
     public final String tagName;
 
     /**
      * Constructs a {@code Tag}.
      *
      * @param tagName A valid tag name.
      */
-    public Tag(String tagName) {
+    private Tag(String tagName) {
         requireNonNull(tagName);
         checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
         this.tagName = tagName;
     }
 
+    /**
+     * Factory method for creating {@code Tag},
+     *
+     * @param tagName A valid tag name.
+     */
+    public static Tag of(String tagName) {
+        if (tagName == null) {
```

Comment:
I think you should do a `requireNonNull(tagName)`

Reply:
thanks